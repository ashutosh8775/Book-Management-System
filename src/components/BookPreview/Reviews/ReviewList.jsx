import React, {Component} from "react";
import StarRatingComponent from 'react-star-rating-component';
import Loader from "react-loader-spinner";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";
import UserIcon from "./user_icon.jpg";
class ReviewList extends Component {
    constructor(props){
        super(props);
    }
    state = {  }
    render() { 
         if (this.props.loading) {
            return <Loader
                        type="ThreeDots"
                        color="#198754"
                        height={80}
                        width={80}
                    />;
        }
        return ( 
           
            <div className="review-list">
                {
                    this.props.reviewList.map(item => {
                        return (
                            <div className="card mb-3" key={item.id}>
                                <div className="card-body">
                                    <div className="user-wrp d-flex align-items-center">
                                        <img src={UserIcon} width="30"/>
                                        <p className="card-subtitle text-muted mb-0">{item.username}</p>
                                    </div>
                                    <div className="rating-wrp d-flex align-items-center">
                                        <StarRatingComponent 
                                            name="rating" 
                                            starCount={5}
                                            value={Number(item.rating)}
                                            editing={false}
                                        />
                                        <h6 className="card-title mb-0">{item.title}</h6>
                                    </div>
                                    <p className="card-text">{item.review}</p>
                                </div>
                            </div>  
                        )  
                    })
                }
            </div>
         );
    }
}
 
export default ReviewList;