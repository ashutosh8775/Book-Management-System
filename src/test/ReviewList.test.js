import React from 'react';
import ReviewList from '../components/BookPreview/Reviews/ReviewList';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('ReviewList Component', ()=>{

    it('List of Reviews', () => {
        const reviewList = [
        { 
            book_id: 3,
            date: "2021-06-18T08:12:32.000Z",
            id: 46,
            rating: 1,
            review: "Not up to the mark",
            title: "Don't Read",
            user_id: 1,
            username: "Ram"
        },
        { 
            book_id: 3,
            date: "2021-06-18T08:12:32.000Z",
            id: 47,
            rating: 2,
            review: "Good Book",
            title: "Awesome",
            user_id: 1,
            username: "test123"
        }
        ];
        const wrapper = shallow(<ReviewList reviewList={reviewList} />);
        expect(wrapper.find('div.card').length).toBe(reviewList.length);
    });
})