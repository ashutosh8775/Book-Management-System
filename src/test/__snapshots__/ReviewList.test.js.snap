// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Component Check Ui is loaded or not 1`] = `ShallowWrapper {}`;
